src/npc.h

```diff
@@ -221,6 +227,7 @@ class Npc final : public Creature
 		std::map<std::string, std::string> m_parameters;
 
 		std::set<Player*> shopPlayerSet;
+		std::set<Player*> spectators;
```

Comment:
It actually depends on implementation, but I think generally set is implemented with a RB tree and you are right. I will change it to an unordered_set as soon as I get home.